src/code/TextTableParser.cs

```diff
@@ -0,0 +1,505 @@
+                    continue;
+                }
+
+                if (Lines.Count == 0) { continue; }
+                // We've collected enough lines to analyze
+                // analyze what we have and then emit them, set the analyzed flag so we will just emit from now on.
+                if (! Analyzed && Lines.Count > WindowSize)
+                {
+                    AnalyzeLines(Lines);
+                    Analyzed = true;
+                    foreach(string l in Lines)
+                    {
+                        Emit(l);
+                    }
+                    Lines.Clear(); // unneeded
+                    // Calculate the column widths
+                }
+            }
+        }
+
+        protected override void EndProcessing()
+        {
+            if (Lines.Count == 0) { return; }
+            if (!Analyzed)
+            {
+                AnalyzeLines(Lines);
+                foreach(string _line in Lines)
+                {
+                    Emit(_line);
+                }
+            }
+        }
+
+        private void Emit(string line)
+        {
+            if (ColumnInfoList != null && columnHeaders != null)
+            {
+                if (AsPsObject)
+                {
+                    WriteObject(GetPsObject(ColumnInfoList, line, columnHeaders));
+                }
+                else
+                {
+                    var jsonOptions = new JsonSerializerOptions();
+                    jsonOptions.MaxDepth = 1;
+                    WriteObject(GetJson(ColumnInfoList, line, columnHeaders));
+                }
+            }
+            else
+            {
+                WriteError(new ErrorRecord(new Exception("No column info"), "NoColumnInfo", ErrorCategory.InvalidOperation, null));
+            }
+        }
+
+        private List<ColumnInfo> ColumnInfoList { get; set; }
+        private string[] columnHeaders { get; set; }
+
+        private void AnalyzeLines(List<string> lines)
+        {
+            if (lines.Count == 0)
+            {
+                return;
+            }
+            SpaceArray = AnalyzeColumns(Lines);
+            spaceRepresentation = GetSpaceRepresentation(Lines.Count, SpaceArray);
+            if (ColumnWidth != null)
+            {
+                ColumnInfoList = GetColumnList(ColumnWidth);
+            }
+            else
+            {
+                ColumnInfoList = GetColumnList(Lines.Count, SpaceArray);
+            }
+
+            if (NoHeader)
+            {
+                columnHeaders = new string[ColumnInfoList.Count];
+                for(int i = 0; i < ColumnInfoList.Count; i++)
+                {
+                    columnHeaders[i] = string.Format("Property_{0:00}", i+1);
+                }
+            }
+            else
+            {
+                columnHeaders = GetHeaderColumns(ColumnInfoList, Lines[HeaderLine]);
+                Lines.RemoveAt(HeaderLine);
+            }
+        }
+
+        /*

```

Comment:
If you're keeping this commented out code, should have a comment explaining why otherwise remove it

Reply:
now that it's in a commit, I don't need it any more